app/models/line_item.rb

```diff
@@ -0,0 +1,38 @@
+class LineItem < ApplicationRecord
+  belongs_to :deal
+  belongs_to :order
+
+  before_save :calculate_net_price
+
+  def calculate_net_price
+    deal = Deal.find(self.deal_id)
```

Comment:
cant we just use deal directly here? instead of finding it again from deal_id

Reply:
yes, as there is a belongs_to association which will return deal object